lib/transaction.js

```diff
@@ -81,12 +98,16 @@ Transaction.prototype.commit = function() {
     .sequelize
     .getQueryInterface()
     .commitTransaction(this, this.options)
-    .finally(function() {
+    .finally(function () {
       self.finished = 'commit';
       if (!self.options.transaction) {
         self.cleanup();
       }
-    });
+    })
+    .tap(
+      function (val) { self.promise.seqResolve(val); },
+      function (err) { self.promise.seqReject(err); }
```

Comment:
So what would i use instead?
There was something that i didn't understand on `promise.js`. If `SequelizePromise` extends promise, why do you create an instance of it instead of proxying it?

instead of:
```js
var SequelizePromise = function (resolver) {
  var self = this;

  var promise = new Promise(function sequelizeResolver(resolve, reject) {
    self.seqResolve = resolve;
    self.seqReject = reject;

    return resolver(resolve, reject);
  });

  promise.seqResolve = this.seqResolve;
  promise.seqReject = this.seqReject;
  promise.$sql = [];

  return promise;
};
```

you could have:
```js
function SequelizePromise ( resolver ) {
  // ride instance of SequelizePromise into Promise's constructor
  Promise.call( this, resolver );

  // whatever you want to do
  ...

  // Since Promise already handles resolver
  return this;
}

Sequelize.prototype.__proto__ = Promise.prototype;
```

Then you could just access `Promise` methods for resolving, rejecting and check promise status.